python/ray/experimental/state.py

```diff
@@ -12,3 +54,198 @@ def get_local_schedulers(worker):
     if client_info[b"client_type"] == b"local_scheduler":
       local_schedulers.append(client_info)
   return local_schedulers
+
+
+class GlobalState(object):
+  """A class used to interface with the Ray control state.
+
+  Attributes:
+    redis_client: The redis client used to query the redis server.
+  """
+  def __init__(self):
+    """Create a GlobalState object."""
+    self.redis_client = None
+
+  def _check_connected(self):
+    """Check that the object has been initialized before it is used.
+
+    Raises:
+      Exception: An exception is raised if ray.init() has not been called yet.
+    """
+    if self.redis_client is None:
+      raise Exception("The ray.global_state API cannot be used before "
+                      "ray.init has been called.")
+
+  def _initialize_global_state(self, redis_ip_address, redis_port):
+    """Initialize the GlobalState object by connecting to Redis.
+
+    Args:
+      redis_ip_address: The IP address of the node that the Redis server lives
+        on.
+      redis_port: The port that the Redis server is listening on.
+    """
+    self.redis_client = redis.StrictRedis(host=redis_ip_address,
+                                          port=redis_port)
+
+  def _object_table(self, object_id_binary):
+    """Fetch and parse the object table information for a single object ID.
+
+    Args:
+      object_id_binary: A string of bytes with the object ID to get information
+        about.
+
+    Returns:
+      A dictionary with information about the object ID in question.
+    """
+    # Return information about a single object ID.
+    object_locations = self.redis_client.execute_command(
+        "RAY.OBJECT_TABLE_LOOKUP", object_id_binary)
+    if object_locations is not None:
+      manager_ids = [binary_to_hex(manager_id)
+                     for manager_id in object_locations]
+    else:
+      manager_ids = None
+
+    result_table_response = self.redis_client.execute_command(
+        "RAY.RESULT_TABLE_LOOKUP", object_id_binary)
+    result_table_message = ResultTableReply.GetRootAsResultTableReply(
+        result_table_response, 0)
+
+    result = {"ManagerIDs": manager_ids,
+              "TaskID": binary_to_hex(result_table_message.TaskId()),
+              "IsPut": bool(result_table_message.IsPut())}
```

Comment:
Since the is being called with a specific object ID, I don't think this is necessary. Note that the user-facing method `object_table` ether takes a specific object ID or returns a dictionary mapping object IDs to the relevant information.